infogami/core/helpers.py

```diff
@@ -22,7 +22,7 @@ def flatten(d):
     {'a': 1, 'b#0': 2, 'b#1': 3, 'c.x': 4, 'c.y': 5}
     """
     def traverse(d, prefix, delim, visit):
-        for k, v in d.iteritems():
+        for k, v in d.items():
```

Comment:
@cclauss  I'm worried this change might cause a performance issue under Python 2, and maybe also in Python 3, I don't _think_ Python 3 `dict.items()` returns an iterator by default? Actually (after some forum searching), I'm not sure about 3, I think the returned `dict_items([])` in 3 will behave like an iterator if used as one... either way, it does change the behaviour under Python 2.

 A more backwards compatible change might be:
`for k, v in iter(d.items()):`   which I _think_ does the same thing in Python 3 and 2.

Thoughts?
If this is a better change, it'd apply to all changed lines in this PR.

Python 2.7 

```
>>> a = {}
>>> a.items()
[]
>>> a.iteritems()
<dictionary-itemiterator object at 0x7f185c096998>
>>> iter(a.iteritems())
<dictionary-itemiterator object at 0x7f185c0969f0>
>>> iter(a.items())
<listiterator object at 0x7f185c0aa050>
```

Python 3
```
>>> a = {}
>>> a.items()
dict_items([])
>>> iter(a.items())
<dict_itemiterator object at 0x7f7cdf156098>
```

Reply:
There are three basic approaches to this:
1. list(a.items())  Slows down Python 3 but is preferred by Python-modernize
2. iter(a.items()) Your suggestion but Python 2 is still going to build the list in the inner block so there is no performance or memory footprint gain and the result can only be used once
3. The way that __six__ does it https://github.com/benjaminp/six/blob/master/six.py#L581

3. is clearly the best and delivers equivalent functionality in both 2 and 3.

Reply:
% __python2.7__
```python
import sys
>>> a = {key: value for value, key in enumerate("abcdefghijklmnopqrstuvwxyz")}
>>> a
{'a': 0, 'c': 2, 'b': 1, 'e': 4, 'd': 3, 'g': 6, 'f': 5, 'i': 8, 'h': 7, 'k': 10, 'j': 9, 'm': 12, 'l': 11, 'o': 14, 'n': 13, 'q': 16, 'p': 15, 's': 18, 'r': 17, 'u': 20, 't': 19, 'w': 22, 'v': 21, 'y': 24, 'x': 23, 'z': 25}
>>> sys.getsizeof(a.items())
280
>>> sys.getsizeof(iter(a.items()))
64
```

Reply:
@cclauss Let's go for 3. and use `six` -- I just canvased @mekarpeles and that is his preference too.

Very good point about my 2. suggestion not making anything better, I clearly didn't think that through properly before typing :)